src/partition/FileUtils.h

```diff
@@ -59,6 +64,36 @@ class InputFile {
     off_t _sz;
 };
 
+class InputFileArrow : public InputFile {
+public:
+    explicit InputFileArrow(boost::filesystem::path const &path, off_t blockSize);
+    ~InputFileArrow();
+
+    /// Return the size of the input file.
+    off_t size() const { return _sz; }
+
+    /// Return the path of the input file.
+    boost::filesystem::path const &path() const { return _path; }
+
+    int getBatchNumber() const;
+
+    /// Read a range of bytes into `buf`.
+    void read(void *buf, off_t off, size_t sz,  int* bufferSize,std::vector<std::string> params) const;
+
+private:
+    // Disable copy construction and assignment.
+    InputFileArrow(InputFileArrow const &);
+    InputFileArrow &operator=(InputFileArrow const &);
```

Comment:
This is done differently in modern C++ (C++11 and later). You need to declare those in the public area of the class as:
```
class InputFileArrow  {
public:
    InputFileArrow(InputFileArrow const &) = delete;
    InputFileArrow &operator=(InputFileArrow const &) = delete;
````

The second comment is that the code formatting is incorrect., You have extra space in `const &`. Should be `const&`.  Please, run this tool on your code before each commit:
```
qserv build -j24 --no-cmake --no-make --no-build-image --clang-format=REFORMAT
```

Reply:
Done